src/utils/disk_utils.hpp

```diff
@@ -25,4 +25,13 @@ inline std::optional<std::string> GetOldDiskKeyOrNull(storage::Delta *head) {
   return std::nullopt;
 }
 
+inline uint64_t GetEarliestTimestamp(storage::Delta *head) {
+  uint64_t ts = head->timestamp->load(std::memory_order_acquire);
+  while (head->next != nullptr) {
+    head = head->next;
```

Comment:
`head = head->next` must be done after reading `head->timestamp` since `next` could be null. Just set ts to max and reorded the lines in the loop.